File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer>ans=new HashMap<>();

        for(int i=0;i<nums.length;i++){
        int x=target-nums[i];

        if(ans.containsKey(x)){
        return new int[]{ans.get(x),i};
        }
        ans.put(nums[i],i);
        }

return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>ans;

        
        

        for(int i=0;i<nums.size();i++){
            int x=target-nums[i];
            if(ans.find(x)!=ans.end()){
                return {i,ans[x]};
            }else{
                ans[nums[i]]=i;
            }
        }
        return {};
    }
};